ospfd/ospf_lsa.c

```diff
@@ -1796,7 +1796,15 @@ struct ospf_lsa *ospf_translated_nssa_originate(struct ospf *ospf,
 		return NULL;
 	}
 
-	extnew = (struct as_external_lsa *)new;
+	extnew = (struct as_external_lsa *)new->data;
```

Comment:
I understand this is the actual fix.  But why do we need to call ospf_lsa_install first?  Why this change?

Reply:
@donaldsharp we need to call `ospf_lsa_install` first so the checksum is computed, otherwise it was still zero. I just ran the test again printing the info before and after `ospf_lsa_install`

```
ep 12 10:26:11 r2 ospfd[23040]:   LSA Header
Sep 12 10:26:11 r2 ospfd[23040]:     LS age 0
Sep 12 10:26:11 r2 ospfd[23040]:     Options 2 (*|-|-|-|-|-|E|-)
Sep 12 10:26:11 r2 ospfd[23040]:     LS type 5 (AS-external-LSA)
Sep 12 10:26:11 r2 ospfd[23040]:     Link State ID 5.5.5.5
Sep 12 10:26:11 r2 ospfd[23040]:     Advertising Router 2.2.2.2
Sep 12 10:26:11 r2 ospfd[23040]:     LS sequence number 0x80000001
Sep 12 10:26:11 r2 ospfd[23040]:     LS checksum 0x0                   <--- ****
Sep 12 10:26:11 r2 ospfd[23040]:     length 36
Sep 12 10:26:11 r2 ospfd[23040]:    Network mask: 32
Sep 12 10:26:11 r2 ospfd[23040]:    Forward addr: 44.44.44.44
Sep 12 10:26:11 r2 ospfd[23040]: ospf_translated_nssa_originate(): translated Type 7, installed:
Sep 12 10:26:11 r2 ospfd[23040]:   LSA Header
Sep 12 10:26:11 r2 ospfd[23040]:     LS age 0
Sep 12 10:26:11 r2 ospfd[23040]:     Options 2 (*|-|-|-|-|-|E|-)
Sep 12 10:26:11 r2 ospfd[23040]:     LS type 5 (AS-external-LSA)
Sep 12 10:26:11 r2 ospfd[23040]:     Link State ID 5.5.5.5
Sep 12 10:26:11 r2 ospfd[23040]:     Advertising Router 2.2.2.2
Sep 12 10:26:11 r2 ospfd[23040]:     LS sequence number 0x80000001
Sep 12 10:26:11 r2 ospfd[23040]:     LS checksum 0x29c8
Sep 12 10:26:11 r2 ospfd[23040]:     length 36
Sep 12 10:26:11 r2 ospfd[23040]:    Network mask: 32
Sep 12 10:26:11 r2 ospfd[23040]:    Forward addr: 44.44.44.44
```
The information displayed by the receiving router:
```
r1# sh ip os database external

       OSPF Router with ID (1.1.1.1)

                AS External Link States

  LS age: 27
  Options: 0x2  : *|-|-|-|-|-|E|-
  LS Flags: 0x6
  LS Type: AS-external-LSA
  Link State ID: 5.5.5.5 (External Network Number)
  Advertising Router: 2.2.2.2
  LS Seq Number: 80000001
  Checksum: 0x29c8
  Length: 36

  Network Mask: /32
        Metric Type: 2 (Larger than any link state path)
        TOS: 0
        Metric: 20
        Forward Address: 44.44.44.44
        External Route Tag: 0
```
Also I realized that most (all?) of the `..._originate` functions like `ospf_external_lsa_originate` the order followed is new, install and dump at the end. Hope it helps.